### PR TITLE
Loss refactor

### PR DESCRIPTION
**Pull request by mikkel**
Refactoring out loss functions to be configurable graph components.

### CODE REVIEW

File: hypergan/losses/standard_gan.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def config():
+    selector = hc.Selector()
+    selector.set("reduce", [tf.reduce_mean])#reduce_sum, reduce_logexp work
+    selector.set("label_smooth", list(np.linspace(0.15, 0.35, num=100)))
+
+    selector.set('create', create)
+
+    return selector.random_config()
+
+def create(config, gan):
```
Comment (martyn) [inline hypergan/losses/standard_gan.py]: This is from improved gan.
